docs/serving/samples/hello-world/helloworld-shell/index.md

```diff
@@ -5,4 +5,206 @@ weight: 1
+   RUN apk update \
+    && apk add netcat-openbsd
+
+   # Copy over the service script
+   COPY script.sh /
+
+   # Start up the webserver
+   CMD [ "/bin/sh", "/script.sh" ]
+   ```
+
+Once you have recreated the sample code files (or used the files in the sample
+folder) you're ready to build and deploy the sample app.
+
+1. Use Docker to build the sample code into a container. To build and push with
+   Docker Hub, run these commands replacing `{username}` with your Docker Hub
+   username:
+
+   ```shell
+   # Build the container on your local machine
+   docker build -t {username}/helloworld-shell .
+
+   # Push the container to docker registry
+   docker push {username}/helloworld-shell
+   ```
+
+## Deploying
+
+1. After the build has completed and the container is pushed to Docker Hub, you
+   can deploy the app into your cluster.
+
+   {{< tabs name="helloworld_shell" default="kn" >}}
+   {{% tab name="yaml" %}}
+
+   1. Create a new file, `service.yaml` and copy the following service definition
+      into the file. Make sure to replace `{username}` with your Docker Hub
+      username.
+
+      ```yaml
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      metadata:
+        name: helloworld-shell
+        namespace: default
+      spec:
+        template:
+          spec:
+            containers:
+              - image: docker.io/{username}/helloworld-shell
+                env:
+                  - name: TARGET
+                    value: "Shell Sample v1"
+      ```
+
+   Ensure that the container image value
+   in `service.yaml` matches the container you built in the previous step. Apply
+   the configuration using `kubectl`:
+
+   ```shell
+   kubectl apply --filename service.yaml
+   ```
+
+   {{< /tab >}}
+   {{% tab name="kn" %}}
+
+   With `kn` you can deploy the service with
+
+   ```shell
+   kn service create helloworld-shell --image=docker.io/{username}/helloworld-shell --env TARGET="Shell Sample v1"
+   ```
+
+   This will wait until your service is deployed and ready, and ultimately it will print the URL through which you can access the service.
+
+   The output will look like:
+
+   ```
+   Creating service 'helloworld-shell' in namespace 'default':
+
+    0.035s The Configuration is still working to reflect the latest desired specification.
+    0.139s The Route is still working to reflect the latest desired specification.
+    0.250s Configuration "helloworld-shell" is waiting for a Revision to become ready.
+    8.040s ...
+    8.136s Ingress has not yet been reconciled.
+    8.277s unsuccessfully observed a new generation
+    8.398s Ready to serve.
+
+  Service 'helloworld-shell' created to latest revision 'helloworld-shell-kwdpt-1' is available at URL:
+  http://helloworld-shell.default.1.2.3.4.xip.io
+  ```
+
+   {{< /tab >}}
+   {{< /tabs >}}
+
+   During the creation of your service, Knative performs the following steps:
+
+   - Create a new immutable revision for this version of the app.
```

Comment:
Wonder if this should be "Creates" since it really is meant to be: "Knative creates ..." and so on, so "Automatically scales ..."

Reply:
good question. I haven't added that part, but maybe the gerund form ("Creating") is a better fit ? (as in the next bullet point "programming" is used, but then we should also change to "scaling" in the third item.

@abrennan89 any recommendation ?

Reply:
"Creation" could be also an option

Reply:
I moved now to "Creates", "Programs the network" and "scales"